### PR TITLE
Improve `hs.data.atomic_resolution_image`

### PR DESCRIPTION
**Pull request by ericpre**
### Progress of the PR
- [x] Add parameters to `hs.data.atomic_resolution_image` to set the size, spacing and rotation of lattice,
- [x] update docstring (if appropriate),
- [n/a] update user guide (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
import hyperspy.api as hs

s = hs.data.atomic_resolution_image(spacing=(0.2, 0.3), rotation_angle=30, size=2048, add_noise=True)
s.plot()
```

### CODE REVIEW

File: hyperspy/data/artificial_data.py [hunk 1]
```diff
@@ -36,9 +44,47 @@
...
+):
     """
-    Get an artificial atomic resolution image.
+    Make an artificial atomic resolution image. The atomic columns
+    are modelled with Gaussian functions.
+
+    Parameters
+    ----------
+    size : int or tuple of int, default=512
+        The number of pixel of the image in horizontal and vertical
```
Comment (jlaehne) [inline hyperspy/data/artificial_data.py]:
```suggestion
        The number of pixels of the image in horizontal and vertical
```

File: hyperspy/data/artificial_data.py [hunk 2]
```diff
@@ -36,9 +44,47 @@
...
+        directions. If int, the size is the same in both directions.
+    spacing : float or tuple of float, default=14
+        The spacing between the atomic columns in horizontal
+        and vertical directions in nanometer.
+    column_radius : float, default=0.05
+        The radius of the atomic column, i. e. the width of the Gaussian
+        in nanometer.
+    rotation_angle : number, default=0
```
Comment (jlaehne) [inline hyperspy/data/artificial_data.py]: `number` -> `int` or `float`?

**Review by jlaehne — COMMENTED** [review]
There are two codecov warnings and one in the documentation build.

File: hyperspy/data/artificial_data.py (hunk 1 above)
Comment (ericpre) [reply hyperspy/data/artificial_data.py]: Done

File: hyperspy/data/artificial_data.py (hunk 2 above)
Comment (ericpre) [reply hyperspy/data/artificial_data.py]: Done

**Comment by jlaehne** [comment]
The docstring says for the `add_noise` parameter "See note to seed the noise to generate reproducible noise." - but there is no note.

**Review by jlaehne — APPROVED** [review]

**Pull request merged**